src/typescript/azure-openapi-validator/rules/XmsIdentifierValidation.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule"
+import { ResourceUtils } from "./utilities/resourceUtils"
+import { deReference } from "./utilities/rules-helper"
+export const XmsIdentifierValidation: string = "XmsIdentifierValidation"
+
+rules.push({
+  id: "R4041",
+  name: XmsIdentifierValidation,
+  severity: "error",
+  category: "SDKViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: ["$.definitions..[?(@.items)]", "$.parameters..[?(@.items)]"],
+  *run(doc, node, path) {
+    const identifiers = node["x-ms-identifiers"] ?? ["id"]
+    const utils = new ResourceUtils(doc)
```

Comment:
the ResourceUtils' constructor will scan all the definitions, could you add an option to prevent this action